hazelcast/src/main/java/com/hazelcast/client/impl/protocol/task/RemoveMigrationListenerMessageTask.java

```diff
@@ -0,0 +1,77 @@
+/*
+ * Copyright (c) 2008-2020, Hazelcast, Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hazelcast.client.impl.protocol.task;
+
+import com.hazelcast.client.impl.protocol.ClientMessage;
+import com.hazelcast.client.impl.protocol.codec.ClientRemoveMigrationListenerCodec;
+import com.hazelcast.instance.impl.Node;
+import com.hazelcast.internal.nio.Connection;
+import com.hazelcast.internal.partition.InternalPartitionService;
+
+import java.security.Permission;
+import java.util.UUID;
+import java.util.concurrent.Future;
+
+public class RemoveMigrationListenerMessageTask
+        extends AbstractRemoveListenerMessageTask<UUID> {
+
+    public RemoveMigrationListenerMessageTask(ClientMessage clientMessage, Node node, Connection connection) {
+        super(clientMessage, node, connection);
+    }
+
+    @Override
+    protected Future<Boolean> deRegisterListener() {
+        InternalPartitionService service = getService(InternalPartitionService.SERVICE_NAME);
+        return service.removeMigrationListenerAsync(parameters);
+    }
+
+    @Override
+    protected UUID getRegistrationId() {
+        return parameters;
+    }
+
+    @Override
+    protected UUID decodeClientMessage(ClientMessage clientMessage) {
+        return ClientRemoveMigrationListenerCodec.decodeRequest(clientMessage);
+    }
+
+    @Override
+    protected ClientMessage encodeResponse(Object response) {
+        return ClientRemoveMigrationListenerCodec.encodeResponse((Boolean) response);
+    }
+
+    @Override
+    public String getServiceName() {
+        return InternalPartitionService.SERVICE_NAME;
+    }
+
+    @Override
+    public Permission getRequiredPermission() {
+        return null;
+    }
+
+    @Override
+    public String getDistributedObjectName() {
+        return null;
```

Comment:
A potentially stupid question for @sancar. Why don't we specify `InternalPartitionService.MIGRATION_EVENT_TOPIC` here, like we do it in `AddMigrationListenerMessageTask`? This way `SecurityContext` interceptor won't be able to handle (and potentially reject) the remove listener operation. Or I'm missing something and there is a reason to have it this way?

Reply:
Marking this one as resolved. The plan is to have a better documentation around `SecurityContext`.